agent/lib/kontena/service_pods/lifecycle_hook_manager.rb

```diff
@@ -0,0 +1,94 @@
+module Kontena
+  module ServicePods
+    class LifecycleHookManager
+      include Kontena::Helpers::RpcHelper
+      include Common
+
+      attr_reader :service_pod
+
+      # @param [ServicePod] service_pod
+      def initialize(service_pod)
+        @service_pod = service_pod
+      end
+
+      # @return [Boolean]
+      def on_pre_start
+        pre_start_hooks = hooks_for('pre_start')
+        return if pre_start_hooks.size == 0
+
+        pre_pod = service_pod.dup
+        pre_pod.entrypoint = '/bin/sh'
+        pre_pod.cmd = ['-c', 'sleep 600']
```

Comment:
So this makes the effective entrypoint to be `/w/w` and cmd will be `['/bin/sh', '-c', 'sleep 600']`. `/bin/sh -c` itself as entrypoint will NOT handle signals properly, not 100% sure if it behaves any better when invoked through `/w/w`. If the sleep does not properly get the signals it will make the containers stop to wait unnecessarily for the `timeout`